### PR TITLE
LPS-147867 Making the upgrade process cluster aware

### PR DESCRIPTION
**Pull request by LuisOrtiz89**
- LPS-147867 New property for controlling the time between retries when obtaining a database lock
- LPS-147867 New property for letting Liferay control de portal startup or rely on the orchestration engine's capacities
- LPS-147867 Adding support to FOR UPDATE
- LPS-147867 New tool for managing database locks when running a piece of code. Intended for controlling the portal startup on clustered environments
- LPS-147867 Applying the locking logic to the portal startup
- LPS-147867 Integration tests
- LPS-147867 Baseline

### CODE REVIEW

**Comment by LuisOrtiz89** [comment]
Hi @shuyangzhou, this is the new PR related to making the upgrade cluster aware. It has changed because now the upgrade of the portal and the upgrade of the modules is not sequential as before, but the locking mechanism is the same.

**Comment by zhangyan0701** [comment]
ci:test:upgrade

**Comment by Kvale1733** [comment]
@LuisOrtiz89 No regressions found.

File: portal-impl/src/com/liferay/portal/tools/DatabaseLockRunner.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+		if (!PropsValues.DATABASE_LOCK_MANAGED_STARTUP) {
+			unsafeRunnable.run();
+
+			return;
+		}
+
+		while (true) {
+			long lockId = _getOrCreateLock(lockKey);
+
+			try (Connection connection = DataAccess.getConnection();
```
Comment (shuyangzhou) [inline portal-impl/src/com/liferay/portal/tools/DatabaseLockRunner.java:53]: @LuisOrtiz89 please do some proper inline to reduce the Connection getting down to 1. There should be no reason to keep getting a new connection.

Comment (LuisOrtiz89) [reply portal-impl/src/com/liferay/portal/tools/DatabaseLockRunner.java]: @shuyangzhou now the class uses just one connection object for all the methods

**Comment by LuisOrtiz89** [comment]
> @LuisOrtiz89 please cleanup the test to remove all timeout logic. CI is unstable on performance, those arbitrary timeout or syncing time would lead to some failures that no one can explain on CI.

@shuyangzhou  Done. All timeout logic has been removed from tests.

**Comment by catarineaguiar** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by LuisOrtiz89** [comment]
Closing the PR as for now this feature will not be implemented. We will review later if it is still needed.

**Pull request closed without merging**